lib/debug/session.rb

```diff
@@ -2030,6 +2035,34 @@ def self.info msg
     log :INFO, msg
   end
 
+  def self.debug msg
+    log :DEBUG, msg
+  end
+
+  # to avoid computing the large number of debug strings unnecessarily
+  DEBUG_MODE = CONFIG[:log_level] == :DEBUG
+
+  def self.debug_event(tc, ev, ev_args, direction:)
+    return unless DEBUG_MODE
+    DEBUGGER__.debug("#{direction} Event (#{ev}): #{ev_args} | #{tc} ")
```

Comment:
@ko1 given that the number of `debug_*` calls can be high I added the `DEBUG_MODE` constant to avoid computing and allocating the log string when it's not in debug mode. do you think it's reasonable or unnecessary?